openssl-dgst, openssl-genpkey: add page (#4879)
tldr-pages/tldr: pages/common/openssl-dgst.md

# openssl dgst

> OpenSSL command to generate digest values and perform signature operations.

> > More information: <https://www.openssl.org/docs/man1.0.2/man1/dgst.html>.

Reviewer: ```suggestion
> More information: <https://www.openssl.org/docs/manmaster/man1/dgst.html>.
```

> > More information: <https://www.openssl.org/docs/manmaster/man1/dgst.html>.

Reviewer: Sorry, I miss with the link :man_facepalming: 
```suggestion
> More information: <https://www.openssl.org/docs/manmaster/man1/openssl-dgst.html>.
```

> - Generate a SHA256 hash value of a file:

Reviewer: Based on `sha256sum` page:
```suggestion
- Calculate the SHA256 checksum for a file, saving the result to a specific file:
```

Author: I don't like the term *checksum* here, because it is usually used only for non-cryptographic algorithms. Would *digest* be OK as well?

Reviewer: Sure

`openssl dgst -sha256 -binary -out {{output_file}} {{input_file}}`

> - Sign a file using and RSA key:

Reviewer: ```suggestion
- Sign a file using an RSA key, saving the result to a specific file:
```

> `openssl dgst -sign {{private_key_file}} -sha256 -sigopt rsa_padding_mode:pss -out {{output_file}} {{input_file}}`

Reviewer: The argument options should be between `{{}}`:
```suggestion
`openssl dgst -sign {{private_key_file}} -sha256 -sigopt {{rsa_padding_mode}}:{{pss}} -out {{output_file}} {{input_file}}`
```

Author: The option parts belong together. I would rather use `{{rsa_padding_mode:pss}}` instead

Reviewer: Is there another possible option for `rsa_padding_mode`? I'm not sure, the man page said `-sigopt nm:v`.

Author: Yes. You can also use `pkcs1` and `none` for signatures.

Reviewer: So, `nm` is an argument option and `v` is another argument option.

Author: No. `nm`is the name of the option and `v` is its value.

Reviewer 2: I think this example is fine as-is.

- Verify an RSA signature:

> `openssl dgst -verify {{public_key_file}} -signature {{signature_file}} -sigopt rsa_padding_mode:pss {{signature_message_file}}`

Reviewer: ```suggestion
`openssl dgst -verify {{public_key_file}} -signature {{signature_file}} -sigopt {{rsa_padding_mode}}:{{pss}} {{signature_message_file}}`
```

Author: The option parts belong together. I would rather use {{rsa_padding_mode:pss}} instead.

Reviewer 2: Same here. I think this example is fine as-is
